mono/metadata/object.c

```diff
@@ -4326,6 +4319,13 @@ mono_unhandled_exception (MonoObject *exc)
 		if (current_appdomain_delegate)
 			call_unhandled_exception_delegate (current_domain, current_appdomain_delegate, exc);
```

Comment:
Unrelated to this PR, but this seems in the wrong order. According to [MSDN](https://msdn.microsoft.com/de-de/library/system.appdomain.unhandledexception(v=vs.110).aspx):

> If the UnhandledException event is handled in the default application domain, it is raised there for any unhandled exception in any thread, no matter what application domain the thread started in.**If the thread started in an application domain that has an event handler for UnhandledException, the event is raised in that application domain**.If that application domain is not the default application domain, and there is also an event handler in the default application domain, the event is raised in both application domains.

We should be calling the current_appdomain_delegate before root_appdomain_delegate